lib/Page.js

```diff
@@ -227,24 +242,65 @@ function getClosestHeading($, headingsSelector, element) {
 }
 
 /**
- * Records headings and keywords inside rendered page into this.headings and this.keywords respectively
+ * Checks if page.frontMatter has a valid page navigation specifier
+ * @returns boolean
  */
-Page.prototype.collectHeadingsAndKeywords = function () {
-  const $ = cheerio.load(fs.readFileSync(this.resultPath));
-  this.collectHeadingsAndKeywordsInContent($(`#${CONTENT_WRAPPER_ID}`).html(), null);
+Page.prototype.validPageNavigationSpecifier = function () {
+  const { pageNav } = this.frontMatter;
+  return pageNav && (pageNav === 'default' || Number.isInteger(pageNav));
 };
 
 /**
- * Generates a heading selector based on the indexing level
- * @param headingIndexingLevel to generate
+ * Collect headings outside of models and panels.
+ * @param content, html content of a page
  */
-function generateHeadingSelector(headingIndexingLevel) {
-  let headingsSelector = 'h1';
-  for (let i = 2; i <= headingIndexingLevel; i += 1) {
-    headingsSelector += `, h${i}`;
+Page.prototype.collectNavigableHeadings = function (content) {
+  if (this.validPageNavigationSpecifier()) {
+    const { pageNav } = this.frontMatter;
+    let elementSelector;
+    // Use specified navigation level or default in this.headingIndexingLevel
+    if (pageNav === 'default') {
+      elementSelector = generateHeadingSelector(this.headingIndexingLevel);
+    } else if (Number.isInteger(pageNav)) {
+      elementSelector = generateHeadingSelector(parseInt(pageNav, 10));
```

Comment:
I believe the radix parameter is optional, so it can be `parseInt(pageNav)` right?

Reply:
> If the input string begins with "0", radix is eight (octal) or 10 (decimal).  Exactly which radix is chosen is implementation-dependent.  ECMAScript 5 specifies that 10 (decimal) is used, but not all browsers support this yet.  **For this reason always specify a radix when using parseInt.** 

Taken from [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt)